File: riveredge-backend/src/apps/haoligo/utils/finance_spec_key.py
"""好力 GO 财务 — 规格比对键（去空格与括号后匹配）。"""

from __future__ import annotations

import re

_SPEC_KEY_STRIP_RE = re.compile(r"[\s()（）\[\]【】{}｛｝]")


def normalize_finance_material_spec_key(spec: str | None) -> str:
    """供应商价格与发票明细规格比对：去掉空格与中英文括号。"""
    if spec is None:
        return ""
    return _SPEC_KEY_STRIP_RE.sub("", str(spec).strip())

File: riveredge-frontend/src/apps/haoligo/utils/financeSpecKey.ts
/** 规格比对键：去掉空格与中英文括号（与后端 finance_spec_key 一致） */

const SPEC_KEY_STRIP_RE = /[\s()（）\[\]【】{}｛｝]/g;

export function normalizeFinanceMaterialSpecKey(spec: unknown): string {
  return String(spec ?? '').trim().replace(SPEC_KEY_STRIP_RE, '');
}
